bin/print_targets.ml

```diff
@@ -0,0 +1,43 @@
+open! Stdune
+open Import
+open Fiber.O
+
+let doc = "Print available targets"
+
+let man =
+  [ `S "DESCRIPTION"
+  ; `P {|$(b,dune targets) print the available targets|}
+  ; `Blocks Common.help_secs
+  ]
+
+let info = Term.info "targets" ~doc ~man
+
+let term =
+  let%map common = Common.term
+  and dirs =
+    Arg.(value
+         & pos_all string []
+         & Arg.info [] ~docv:"DIRS")
+  in
+  Common.set_common common ~targets:[] ;
+  let log = Log.create common in
+  Scheduler.go ~log ~common (fun () ->
+      Import.Main.setup ~log common ~external_lib_deps_mode:true
+      >>= fun setup ->
+      let files = Build_system.all_targets setup.build_system in
+      let files =
+        if List.is_empty dirs then files
+        else
+          let in_dir x dir =
+            let of_ = Path.of_string ("_build/default/" ^ dir) in
+            Path.is_descendant x ~of_
+          in
+          let in_dirs x = List.exists ~f:(in_dir x) in
+          List.filter files ~f:(fun x -> in_dirs x dirs)
```

Comment:
This is similar to the `target_hint` function in `target.ml`. Could you refactor the code to use the same function between here and `target_hint`? In particular, the code in `target_hint` is a bit more generic as it works with any build context.

Reply:
I deleted the DIRS parameter until I figure out how to get the targets inside of these directories as well.